### PR TITLE
Icon resize

### PR DESCRIPTION
**Pull request by jeabakker**
fixed several problems with image resizing

ref #10686

### CODE REVIEW

File: engine/classes/Elgg/ImageService.php [hunk 1]
```diff
@@ -198,4 +199,42 @@ public function normalizeResizeParameters($width, $height, array $params = []) {
...
+	/**
+	 * Determine the image file format, this is needed for correct resizing
+	 *
+	 * @param string $filename path to the file
+	 * @param array  $params   array of resizing params (can contain 'format' to set save format)
+	 *
+	 * @see https://github.com/Elgg/Elgg/issues/10686
+	 * @return void|string
+	 */
+	protected function getFileFormat($filename, $params) {
```
Comment (hypeJunction) [inline engine/classes/Elgg/ImageService.php]: Can we use Imagine to do this? We also have a similar method in one of the file related classes (see ElggFile::detectMimeType())

Comment (jeabakker) [inline engine/classes/Elgg/ImageService.php]: > Can we use Imagine to do this? 

I looked but couldn't find any

>We also have a similar method in one of the file related classes (see ElggFile::detectMimeType())

maybe but `$filename` doens't have to be in the Elgg file structure (eg uploaded file is in /tmp)

Comment (hypeJunction) [inline engine/classes/Elgg/ImageService.php]: See `\Elgg\Filesystem\MimeTypeDetector`, it's agnostic to Elgg's file system and works with file paths.

Comment (jeabakker) [inline engine/classes/Elgg/ImageService.php]: > See \Elgg\Filesystem\MimeTypeDetector, it's agnostic to Elgg's file system and works with file paths.

I changed it, thanks for the tip

**Comment by hypeJunction** [comment]
Can we add a test to see what the expected outcome is? From the code it's unclear what issue this is fixing.

**Comment by jeabakker** [comment]
> Can we add a test to see what the expected outcome is? From the code it's unclear what issue this is fixing.

I added 3 tests for resizing
1. resize from an image with an image extension
2. resize from a temp file (extension is `.tmp`, but the contents is still an image)
3. resize from an image without an extension

**Review by hypeJunction — APPROVED** [review]

**Pull request merged**